### PR TITLE
default some more

### PR DESCRIPTION
**Pull request by neheb**
swap is not needed here as shared_ptr is copyable.

### CODE REVIEW

File: taglib/mp4/mp4coverart.cpp [hunk 1]
```diff
@@ -47,14 +47,8 @@ MP4::CoverArt::CoverArt(Format format, const ByteVector &data) :
...
 
-MP4::CoverArt::CoverArt(const CoverArt &item) = default;
-
-MP4::CoverArt &
-MP4::CoverArt::operator=(const CoverArt &item)
-{
-  CoverArt(item).swap(*this);
-  return *this;
-}
+MP4::CoverArt::CoverArt(const CoverArt &)                 = default;
```
Comment (ufleisch) [inline taglib/mp4/mp4coverart.cpp:50]:
```suggestion
MP4::CoverArt::CoverArt(const CoverArt &) = default;
```

File: taglib/mp4/mp4item.cpp [hunk 2]
```diff
@@ -53,14 +53,8 @@ MP4::Item::Item() :
...
 
-MP4::Item::Item(const Item &item) = default;
-
-MP4::Item &
-MP4::Item::operator=(const Item &item)
-{
-  Item(item).swap(*this);
-  return *this;
-}
+MP4::Item::Item(const Item &)                 = default;
```
Comment (ufleisch) [inline taglib/mp4/mp4item.cpp:56]:
```suggestion
MP4::Item::Item(const Item &) = default;
```

Comment (neheb) [reply taglib/mp4/mp4item.cpp]: Fixed.

File: taglib/mp4/mp4coverart.cpp (hunk 1 above)
Comment (neheb) [reply taglib/mp4/mp4coverart.cpp]: Fixed.

**Pull request merged**